### PR TITLE
Async Await Refactor

### PR DESCRIPTION
**Pull request by johnktravers**
- Refactor new favorite location endpoint using async/await syntax
- Add happy and sad path testing for new favorite endpoint (invalid/missing api key and/or location)
- Overall test coverage is at 67%

I'm going to clean the router file up by creating a service object and moving the helper methods into a module. @rhantak do you see any other ways that the `router.post` method could be refactored?

### CODE REVIEW

File: routes/api/v1/favorites.js [hunk 1]
```diff
@@ -10,38 +10,57 @@ const { URL, URLSearchParams } = require('url');
 
 const Geocode = require('../../../lib/objects/geocode');
 
-router.post('/', function(req, res, next) {
-  let location = req.body.location;
-  let apiKey = req.body.api_key;
+router.post('/', async function(req, res, next) {
+  let location = req.body.location || '';
+  let apiKey = req.body.api_key || '';
```
Comment (rhantak) [inline routes/api/v1/favorites.js]: I like that these are memoized, that's a great catch for improperly formatted request bodies.

**Comment by rhantak** [comment]
I think overall this is great, it makes the code much more modular, which definitely helps when adding new functionality moving forward (like listing favorites). The only issue I see is that Travis CI is failing, looks like it's receiving a 400 status where it expects a 200 on your test for creating a new favorite.

**Comment by johnktravers** [comment]
The tests are passing locally, it's only on Travis CI that the first test fails. I think you're right about the environment variables not being set up correctly on Travis. I'll double check that and hopefully everything will pass!

**Comment by rhantak** [comment]
Everything looks good now! Must've been it.

**Pull request merged**